docs/operating-eck/restrict-cross-namespace-associations.asciidoc

```diff
@@ -0,0 +1,94 @@
+:page_id: restrict-cross-namespace-associations
+ifdef::env-github[]
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/restrict-cross-namespace-associations.html[View this document on the Elastic website]
+****
+endif::[]
+[id="{p}-{page_id}"]
+= Restrict cross namespace resource associations
+
+By default, associations can be created between any resources as long as they are deployed in namespaces managed by the same ECK instance.
+
+This is true, for example, when using the `elasticsearchRef` field to automatically configure and establish a trust relationship between Kibana and Elasticsearch. If a user can create a Kibana instance in a namespace managed by ECK, then this Kibana instance can be associated with any Elasticsearch cluster deployed in another namespace managed by the same ECK instance, even if the user only has permissions to edit Kibana and not the Elasticsearch resource.
+
+This section describes how you can restrict the associations that can be created between resources managed by ECK.
+
+== Enabling cross namespace association restriction
+
+This feature is disabled by default. To enable it, start the operator with the `--enforce-rbac-on-refs` flag.
+
+NOTE: This feature only provides an access control for resources deployed across two different namespaces. Associations can still be freely created between resources deployed in the same namespace.
+
+Once enabled, associations can be created as long as the `ServiceAccount` specified in the associated resources can execute an HTTP `get` request against the referenced Elasticsearch object.
+
+IMPORTANT: ECK removes any association that is not allowed. If you have existing associations, do not enable this feature without creating the required `Roles` and `RoleBindings` as described in the following sections.
+
+== Allowing associations to be created
+
+First create a `ServiceAccount`:
+
+[source,sh]
+----
+> kubectl create serviceaccount associated-resource-sa
+----
+
```

Comment:
I think that it might be better to move the section on creating the cluster role and associating it with the service account here because those are one-off tasks that are closely related.